### PR TITLE
6 core get api articles article id comments

### PR DESCRIPTION
**Pull request by mikef80**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,6 +47,56 @@ describe("/api/topics", () => {
   });
 });
 
+describe.only("/api/articles/:article_id/comments", () => {
```
Comment (nc-verity-gregory) [inline __tests__/app.test.js:50]: Please don't leave .only in your pr code. Please remove and resubmit.

**Review by nc-verity-gregory — CHANGES_REQUESTED** [review]
Please don't leave .only in your test file. I know its large and will save time for you to run it, however it gets past the husky pre commit hook and we need to know all tests are passing. Please make sure they are then resubmit.

File: __tests__/app.test.js (hunk 1 above)
Comment (mikef80) [reply __tests__/app.test.js:50]: Bugger - sorry, that was unintentional.  Will resubmit asap

File: __tests__/app.test.js [hunk 2]
```diff
@@ -47,6 +47,56 @@ describe("/api/topics", () => {
...
+      });
+  });
+
+  it("GET:200 returns elements that each have the correct properties and sorted by most recent first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.comments.forEach((comment, index) => {
+          expect(comment).toContainKeys([
```
Comment (nc-verity-gregory) [inline __tests__/app.test.js:67]: We could also assert what the values look like as well, how do we know we are getting comments with article_id of 1?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -47,6 +47,56 @@ describe("/api/topics", () => {
...
+      });
+  });
+
+  it("GET:400 responds with 'bad request' when passed an invalid article_id", () => {
+    return request(app)
+      .get("/api/articles/chickens/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
```
Comment (nc-verity-gregory) [inline __tests__/app.test.js:96]: Cool, what happens if we ask for comments on article 2? It currently doesn't have any comments, what status code would that be?

File: api/models/comments-model.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const db = require("../../db/connection");
+
+exports.selectCommentsByArticleId = (article_id) => {
+  return db
+    .query(`SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;`, [article_id])
+    .then((results) => {
+      if (!results.rows.length) {
```
Comment (nc-verity-gregory) [inline api/models/comments-model.js:7]: Is it a 404 if there are no comments on an article that does exist?

**Review by nc-verity-gregory — CHANGES_REQUESTED** [review]
Just one more test to look at. Good effort so far, I'd like to see that final test for a valid article that has no comments :)

File: api/models/comments-models.js [hunk 5]
```diff
@@ -3,10 +3,7 @@ const db = require("../../db/connection");
 exports.selectCommentsByArticleId = (article_id) => {
   return db
     .query(`SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;`, [article_id])
-    .then((results) => {
-      if (!results.rows.length) {
-        return Promise.reject({ status: 404, msg: "not found" });
-      }
-      return results.rows;
+    .then(({rows}) => {
+      return rows;
```
Comment (nc-mick-fay) [inline api/models/comments-models.js:7]: Good change to send back an empty array rather than having an error!

**Review by nc-mick-fay — APPROVED** [review]
After the addition of this test and these changes, I'm happy to approve this pull request. 
Merge into main when you're ready

**Pull request merged**